staging/src/k8s.io/client-go/rest/with_retry.go

```diff
@@ -245,6 +283,34 @@ func (r *withRetry) After(ctx context.Context, request *Request, resp *http.Resp
 	}
 }
 
+func (r *withRetry) WrapPreviousError(finalErr error) error {
+	if finalErr == nil || r.previousErr == nil {
+		return finalErr
+	}
+	if !isContextError(finalErr) {
+		return finalErr
+	}
+
+	previousErr := r.previousErr
+	if err := errors.Unwrap(r.previousErr); err != nil {
+		previousErr = err
+	}
+
+	return fmt.Errorf("%w - error from a previous attempt: %s", finalErr, previousErr.Error())
```

Comment:
If you have both a current and a previous, why would this method not include both?

Reply:
I don't think we can create an error chain with both `current` and `previous` errors being immutable to us - we can wrap current error in the error chain so behaviors from `errors.Is` and `errors.Unwrap` are consistent.

This PR achieves the following
```
err := WrapPreviousError(currentErr)

errors.Is(err, currentErr) is true
errors.Is(err, previousErr) is false
```

`errors.Unwrap` must yield `currentErr` to maintain backward compatibility. If that's the case, then from `currentErr` there is no way for us to go to `previousErr` since `previousErr` is from a different error chain, also we can't mutate `currentErr`. Thoughts?

Reply:
to clarify, i am including both:
- the current error is wrapped
- the wrapped error provides the message of the previous error